### PR TITLE
feat: add BitVec.toInt_sub, simplify BitVec.toInt_neg

### PR DESCRIPTION
**Pull request by tobiasgrosser**
This also requires us to expand the theory of `Int.bmod`.

### CODE REVIEW

File: src/Init/Data/Int/Lemmas.lean [hunk 1]
```diff
@@ -54,8 +63,32 @@ theorem negOfNat_eq : negOfNat n = -ofNat n := rfl
...
+    cases n <;> simp_all
+  case h_2 ia ib na nb ha hb =>
+    have h' := eq_add_one_iff_neg_eq_negSucc.mpr hb
+    have h'' := (@ofNat_inj m na).mp ha
+    rw [h'] at h
+    rw [h''] at h
+    rw [subNatNat_of_sub h]
+    congr <;> simp_all
+  · simp_all
+  · simp_all
```
Comment (tobiasgrosser) [inline src/Init/Data/Int/Lemmas.lean:90]: I feel the proof above could benefit from some golfing.

File: src/Init/Data/BitVec/Lemmas.lean [hunk 2]
```diff
@@ -1974,6 +1980,10 @@ theorem sub_def {n} (x y : BitVec n) : x - y = .ofNat n ((2^n - y.toNat) + x.toN
 @[simp] theorem toNat_sub {n} (x y : BitVec n) :
     (x - y).toNat = (((2^n - y.toNat) + x.toNat) % 2^n) := rfl
 
+@[simp, bv_toNat] theorem toInt_sub {x y : BitVec w} :
+  (x - y).toInt = (x.toInt - y.toInt).bmod (2^w) := by
```
Comment (alexkeizer) [inline src/Init/Data/BitVec/Lemmas.lean:1984]:
```suggestion
    (x - y).toInt = (x.toInt - y.toInt).bmod (2 ^ w) := by
```

**Comment by alexkeizer** [comment]
With `toInt_sub` done we can also simplify the proof of `toInt_neg`, I just pushed a commit that did just that

**Comment by tobiasgrosser** [comment]
@kim-em, your feedback would be appreciated especially with respect to the `Int` changes.

**Pull request merged**